### PR TITLE
Cleanup cstlye render_const [run_process_replay]

### PR DESCRIPTION
**Pull request by greg-niemeyer**
Simplifies cstyle `render_const` in the spirit of #5313. This change takes advantage of the fact that CONSTs are always scalar, and localizes the complexity of deciding when to apply casts and extra parentheses to CONSTs in ALUs, where CONSTs are used to construct expressions as opposed to being used for simple assignments (where type promotion is easy to reason about).

process_replay impact: less excessive casting and parens.

e.g.
image

### CODE REVIEW

**Comment by Qazalin** [comment]
because NAN is always float32 this should cast in render_const.
#define NAN (__int_as_float(0x7fffffff))"
In general, I'm fine with doing all render_casts in the const renderer.

**Comment by geohot** [comment]
Can rendering vectorized just be removed from render const as a separate PR?

**Comment by geohot** [comment]
Maybe assert its count is 1

File: tinygrad/renderer/cstyle.py [hunk 1]
```diff
@@ -45,13 +45,13 @@ def render_vectorize(self, x:List[str], var_dtype:DType) -> str:
 
   # returns a str expression of the const with the given type
   def render_const(self, x:ConstType, dtype:DType) -> str:
+    assert dtype.count == 1, "consts should be scalar, got {dtype}"
```
Comment (Qazalin) [inline tinygrad/renderer/cstyle.py:48]: err

**Pull request merged**